firefox-ios/firefox-ios-tests/Tests/ClientTests/Sharing/ShareTelemetryTests.swift

```diff
@@ -94,17 +115,29 @@ final class ShareTelemetryTests: XCTestCase {
             isOptedInSentFromFirefox: testIsOptedInSentFromFirefox
         )
 
-        testEventMetricRecordingSuccess(metric: GleanMetrics.ShareSheet.sharedTo)
+        let savedEvent = try XCTUnwrap(
+            gleanWrapper.savedEvent as? EventMetricType<GleanMetrics.ShareSheet.SharedToExtra>
+        )
+        let savedExtras = try XCTUnwrap(
+            gleanWrapper.savedExtras as? GleanMetrics.ShareSheet.SharedToExtra
+        )
+
+        let extraRecord = savedExtras.toExtraRecord()
+
+        let expectedMetricType = type(of: GleanMetrics.ShareSheet.sharedTo)
+        let resultMetricType = type(of: savedEvent)
 
-        let resultValue = try XCTUnwrap(GleanMetrics.ShareSheet.sharedTo.testGetValue())
-        XCTAssertEqual(resultValue[0].extra?[activityIdentifierKey], testActivityType.rawValue)
-        XCTAssertEqual(resultValue[0].extra?[shareTypeKey], testShareType.typeName)
-        XCTAssertEqual(resultValue[0].extra?[hasShareMessageKey], String(testHasShareMessage))
-        XCTAssertEqual(resultValue[0].extra?[hasIsEnrolledInSentFromFirefoxKey], String(testIsEnrolledInSentFromFirefox))
-        XCTAssertEqual(resultValue[0].extra?[hasIsOptedInSentFromFirefoxKey], String(testIsOptedInSentFromFirefox))
+        let message = TelemetryDebugMessage(expectedMetric: expectedMetricType, resultMetric: resultMetricType)
+        XCTAssert(resultMetricType == expectedMetricType, message.text)
+        XCTAssertEqual(gleanWrapper.recordEventCalled, 1)
+        XCTAssertEqual(extraRecord[activityIdentifierKey], testActivityType.rawValue)
+        XCTAssertEqual(extraRecord[shareTypeKey], testShareType.typeName)
+        XCTAssertEqual(extraRecord[hasShareMessageKey], String(testHasShareMessage))
+        XCTAssertEqual(extraRecord[hasIsEnrolledInSentFromFirefoxKey], String(testIsEnrolledInSentFromFirefox))
+        XCTAssertEqual(extraRecord[hasIsOptedInSentFromFirefoxKey], String(testIsOptedInSentFromFirefox))
     }
 
     func createSubject() -> ShareTelemetry {
-        return DefaultShareTelemetry()
+        return DefaultShareTelemetry(gleanWrapper: GleanWrapper)
```

Comment:
```suggestion
        return DefaultShareTelemetry(gleanWrapper: gleanWrapper)
```
Just noticed after after the fact, we should be passing in the gleanWrapper variable rather than the protocol itself :)